src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -1722,6 +1723,51 @@ def _public_key_bytes(self, encoding, format, evp_pkey, cdata):
                 write_bio = self._lib.i2d_RSAPublicKey_bio
 
             key = cdata
+        elif format is serialization.PublicFormat.OpenSSH:
+            if encoding is not serialization.Encoding.OpenSSH:
+                raise ValueError(
+                    "OpenSSH format must be used with OpenSSH encoding"
+                )
+            if isinstance(key, rsa.RSAPublicKey):
```

Comment:
The OpenSSH branch makes this function really long and a bit hard to read. What do you think about splitting this into a separate method? It will obviously only be called once but would be a method that takes a key and returns a bytestring.